pkg/load/agents/configAgent.go

```diff
@@ -239,5 +263,89 @@ func (a *configAgent) buildIndexes() {
 				}
 			}
 		}
+
+		// Building the diff is expensive, so cache it in case we have multiple
+		// subscribers.
+		var changes []IndexDelta
+		for _, channel := range a.indexSubscribers[indexName] {
+			if changes == nil {
+				changes = buildIndexDelta(oldIndexes[indexName], a.indexes[indexName])
+			}
+
+			// This might block, so do it in a new goroutine
+			channel := channel
```

Comment:
nit: totally a style thing but I think making this a named parameter of the anonymous func in the routine and passing it is a little more explicit as to what the code is doing

```go
func(channel chan <-IndexDelta) {
    //...
}(channel)
```

Reply:
I hate that notation because you can not read the code from top to bottom anymore, you have to go to the start of the closure, then to the end to see what  arguments it gets, then to the start again to read it.

This notation here allows to simply read the code from top to bottom.